src/sfizz/Synth.cpp

```diff
@@ -1296,7 +1303,16 @@ void Synth::timePosition(int delay, int bar, double barBeat)
     Impl& impl = *impl_;
     ScopedTiming logger { impl.dispatchDuration_, ScopedTiming::Operation::addToDuration };
 
-    impl.resources_.beatClock.setTimePosition(delay, BBT(bar, barBeat));
+    const auto newPosition = BBT(bar, barBeat);
+    const auto newBeatPosition = newPosition.toBeats(impl.resources_.beatClock.getTimeSignature());
+    const auto currentBeatPosition = impl.resources_.beatClock.getLastBeatPosition();
+    const auto positionDifference = std::abs(newBeatPosition - currentBeatPosition);
+    const auto threshold = 2 * static_cast<float>(impl.resources_.beatClock.getBeatsPerFrame());
+
+    if (positionDifference > threshold)
+        impl.playheadMoved_ = true;
```

Comment:
This logic is not super pretty although it does make some sense. There may be a better way to assess whether the playhead moved.

Reply:
Perhaps reset them only at T=0 ?
It seems working in Reaper to check for bar=0 && beat=0 && playing=on.

Reply:
Yeah I was talking about the detection of the moved playhead. For the reset conditions there are many variations. It seems sforzando does not reset only on playhead moved but I'm not sure it's so natural.

Reply:
In Alex's example if you have the ability to render/bounce a region, only resetting at T=0 would smooth if the region you render does not start on 0.